src/cargo/core/registry.rs

```diff
@@ -93,7 +97,8 @@ impl<'cfg> PackageRegistry<'cfg> {
         let mut ret = Vec::new();
 
         for (_, source) in self.sources.sources_mut() {
-            try!(source.download(package_ids));
+            try!(network::with_retry(&self.config, || source.download(package_ids)));
```

Comment:
This probably wants to be attached to the literal network calls themselves rather than way up here at the `download` layer. For example a download from everything but the registry doesn't actually need to be retried.